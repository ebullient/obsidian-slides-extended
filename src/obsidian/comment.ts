import { Properties } from "./transformers";

export class Comment extends Properties {
    type: string;

    constructor(type: string, attributes: Map<string, string>) {
        attributes.set("onTarget", type);
        super(attributes);
        this.type = type;
    }

    public static of(
        type: string,
        style: string[] = [],
        clazz: string[] = [],
        attributes: Map<string, string> = new Map<string, string>(),
    ): Comment {
        if (clazz && clazz.length > 0) {
            const classValue = attributes.get("class");

            if (classValue) {
                const split = classValue.split(" ");
                split.push(...clazz);
                attributes.set("class", split.join(" "));
            } else {
                attributes.set("class", clazz.join(" "));
            }
        }

        if (style && style.length > 0) {
            const styleValue = attributes.get("style");

            if (styleValue) {
                const split = styleValue
                    .split(";")
                    .map((value) => value.trim());
                split.push(...style);
                attributes.set("style", split.join("; "));
            } else {
                attributes.set("style", style.join("; "));
            }
        }

        return new Comment(type, attributes);
    }
}

export class CommentParser {
    private readCommentRegex = /<!--.*?-->/s;
    private parseRegex = /<!--\s*(?:\.)?(element|slide):?\s*(.*)-->/s;
    private parsePropertiesRegex =
        /([^=]*)\s*=\s*"([^"]*)"\s*|([^=]*)\s*=\s*'([^']*)'\s*|([^ ]*)\s*/g;

    commentToString(comment: Comment): string {
        return `<!-- .${comment.type}: ${this.buildAttributes(comment)} -->`;
    }

    buildAttributes(comment: Comment): string {
        const styles = comment.getStyles();
        const classes = comment.getClasses();

        const stylesString = styles.length > 0 ? `style="${styles}" ` : "";
        const classesString = classes.length > 0 ? `class="${classes}" ` : "";

        return `${stylesString}${classesString}${comment.getAttributes()}`.trim();
    }

    parseLine(line: string): Comment {
        return this.lineHasComment(line)
            ? this.parseComment(this.readCommentStringFromLine(line))
            : null;
    }

    parseComment(comment: string): Comment {
        try {
            const [, type, properties] = this.parseRegex.exec(comment);
            const attributes = this.parseAttributes(properties);

            return new Comment(type, attributes);
        } catch (_ex) {
            console.error(`ERROR: Cannot parse comment: ${comment}`);
            return null;
        }
    }

    buildComment(
        type: string,
        style: string[] = [],
        clazz: string[] = [],
        attributes: Map<string, string> = new Map<string, string>(),
    ): Comment {
        return Comment.of(type, style, clazz, attributes);
    }

    lineHasComment(line: string): boolean {
        return this.parseRegex.test(line);
    }

    replace(line: string, comment: Comment): string {
        return line.replace(
            this.readCommentRegex,
            this.commentToString(comment),
        );
    }

    private readCommentStringFromLine(line: string): string {
        return this.readCommentRegex.exec(line)?.[0] ?? "";
    }

    private parseAttributes(properties: string): Map<string, string> {
        const attributes = new Map<string, string>();

        while (true) {
            const m = this.parsePropertiesRegex.exec(properties);
            if (m == null) {
                break;
            }
            if (m.index === this.parsePropertiesRegex.lastIndex) {
                this.parsePropertiesRegex.lastIndex++;
            }

            let key: string;
            let value: string;

            m.forEach((match, groupIndex) => {
                if (groupIndex === 1 || groupIndex === 3) {
                    key = match;
                }
                if (groupIndex === 2 || groupIndex === 4) {
                    value = match;
                    attributes.set(key, value);
                }
                if (groupIndex === 5) {
                    if (match) {
                        attributes.set(match, "true");
                    }
                }
            });
        }

        attributes.delete(undefined);
        return attributes;
    }
}
