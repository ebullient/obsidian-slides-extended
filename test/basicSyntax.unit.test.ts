import { MarkdownProcessor } from 'src/obsidian/markdownProcessor';
import { when } from 'ts-mockito';
import { MockedObsidianUtils, obsidianUtils as utilsInstance } from './__mocks__/mockObsidianUtils';
import { prepare } from './testUtils';
import { readFileSync } from 'node:fs';

test('Basic Markdown Syntax > Headers', () => {
    const input = `# This is a heading 1
## This is a heading 2
### This is a heading 3
#### This is a heading 4
##### This is a heading 5
###### This is a heading 6`;

    const { options, markdown } = prepare(input);
    const sut = new MarkdownProcessor(utilsInstance);

    const result = JSON.stringify(sut.process(markdown, options));
    return expect(result).toMatchSnapshot();
});

test('Basic Markdown Syntax > Text style', () => {
    const input = `*This text will be italic*

_This will also be italic_

**This text will be bold**

__This will also be bold__

%%This is a Comment(Can't see it)%%

_You **can** combine them_

---

Any word wrapped with two tildes (like ~~this~~) will appear crossed out.

Any word wrapped with two equal signs (like ==this==) will appear as highlighted.`;

    const { options, markdown } = prepare(input);
    const sut = new MarkdownProcessor(utilsInstance);

    const result = JSON.stringify(sut.process(markdown, options));
    return expect(result).toMatchSnapshot();
});

test('Basic Markdown Syntax > Lists', () => {
    const input = `- Item 1
- Item 2
	- Item 2a
	- Item 2b

---

1. Item 1
1. Item 2
1. Item 3
   1. Item 3a
   1. Item 3b
`;

    const { options, markdown } = prepare(input);
    const sut = new MarkdownProcessor(utilsInstance);

    const result = JSON.stringify(sut.process(markdown, options));
    return expect(result).toMatchSnapshot();
});

test('Basic Markdown Syntax > Images', () => {
    when(MockedObsidianUtils.findMediaFile('Image.jpg')).thenCall(arg => {
        return '/documentation/Image.jpg';
    });

    when(MockedObsidianUtils.findMediaFile('/local-file-url/Users/testUser/Desktop/howToUse.png')).thenCall(arg => {
        return '/local-file-url/Users/testUser/Desktop/howToUse.png';
    });

    const input = `
Insert image with standard markdown syntax

![Image](https://picsum.photos/id/1006/500/300)

---

Insert image that lies outside of your vault

![Image](file:///Users/testUser/Desktop/howToUse.png)

---

Insert image with obsidian markdown syntax

![[Image.jpg]]

---

Scale image to a width of 100 px

![[Image.jpg|100]]

---

Scale image to a width of 300x100 px

![[Image.jpg|300x100]] <!-- element style="object-fit: cover" -->

`;

    const { options, markdown } = prepare(input);
    const sut = new MarkdownProcessor(utilsInstance);

    const result = JSON.stringify(sut.process(markdown, options));
    return expect(result).toMatchSnapshot();
});

test('Basic Markdown Syntax > Videos', () => {
    when(MockedObsidianUtils.findMediaFile('/local-file-url/Users/testUser/Desktop/video.mp4')).thenCall(arg => {
        return '/local-file-url/Users/testUser/Desktop/video.mp4';
    });

    when(MockedObsidianUtils.findMediaFile('video.mp4')).thenCall(arg => {
        return '/documentation/video.mp4';
    });

    const input = `
Insert image that lies outside of your vault

![Video](file:///Users/testUser/Desktop/video.mp4)

---

Insert image with obsidian markdown syntax

![[video.mp4]]

---

Scale image to a width of 100 px

![[video.mp4|100]]

---

Scale image to a width of 300x100 px, disable controls

![[video.mp4|300x100]] <!-- element style="object-fit: cover" controls="false" -->

`;

    const { options, markdown } = prepare(input);
    const sut = new MarkdownProcessor(utilsInstance);

    const result = JSON.stringify(sut.process(markdown, options));
    return expect(result).toMatchSnapshot();
});


test('Basic Markdown Syntax > Links', () => {
    const input = `External Links

http://obsidian.md - automatic!

[Obsidian](http://obsidian.md)

---

Obsidian URI links

[Link to note](obsidian://open?path=D:%2Fpath%2Fto%2Ffile.md)

[Link to note](obsidian://open?vault=MainVault&file=MyNote.md)

---

Internal Links

This [[Internal link]] will be displayed as normal text

This [[Internal link|Link]] will use its alias for displaying

`;

    const { options, markdown } = prepare(input);
    const sut = new MarkdownProcessor(utilsInstance);

    const result = JSON.stringify(sut.process(markdown, options));
    return expect(result).toMatchSnapshot();
});

test('Basic Markdown Syntax > Links', () => {
    const input = `---
enableLinks: true
---

Internal Links

This [[Internal link]] will be displayed as normal text

This [[Internal link|Link]] will use its alias for displaying

`;

    const { options, markdown } = prepare(input);
    const sut = new MarkdownProcessor(utilsInstance);

    const result = JSON.stringify(sut.process(markdown, options));
    return expect(result).toMatchSnapshot();
});

test('Basic Markdown Syntax > Embeds', () => {

    when(MockedObsidianUtils.parseFile('Obsidian Notes.md', 'Link')).thenCall(arg => {
        return 'Link to Obsidian Homepage: http://obsidian.md';
    });

    when(MockedObsidianUtils.parseFile('Obsidian Notes.md', null)).thenCall(arg => {
        return `# Notes about Obsidian
A knowledge base that works on local Markdown files

# Link
Link to Obsidian Homepage: http://obsidian.md`;
    });

    const input = `![[Obsidian Notes|Title]]

---

![[Obsidian Notes#Link]]

---

![Title](Obsidian%20Notes.md)

---

![](Obsidian%20Notes.md#Link)
`;

    const { options, markdown } = prepare(input);
    const sut = new MarkdownProcessor(utilsInstance);

    const result = JSON.stringify(sut.process(markdown, options));
    return expect(result).toMatchSnapshot();
});

test('Basic Markdown Syntax > Blockquotes', () => {
    const input = `> Human beings face ever more complex and urgent problems, and their effectiveness in dealing with these problems is a matter that is critical to the stability and continued progress of society.

- Doug Engelbart, 1961
`;

    const { options, markdown } = prepare(input);
    const sut = new MarkdownProcessor(utilsInstance);

    const result = JSON.stringify(sut.process(markdown, options));
    return expect(result).toMatchSnapshot();
});

test('Basic Markdown Syntax > Inline Code', () => {
    const input = `### Text inside \`backticks\` on a line will be formatted like code.
`;

    const { options, markdown } = prepare(input);
    const sut = new MarkdownProcessor(utilsInstance);

    const result = JSON.stringify(sut.process(markdown, options));
    return expect(result).toMatchSnapshot();
});

test('Basic Markdown Syntax > Tables', () => {
    const input = `First Header | Second Header
------------ | ------------
Content from cell 1 | Content from cell 2
Content in the first column | Content in the second column
`;

    const { options, markdown } = prepare(input);
    const sut = new MarkdownProcessor(utilsInstance);

    const result = JSON.stringify(sut.process(markdown, options));
    return expect(result).toMatchSnapshot();
});

test('Basic Markdown Syntax > Footnotes', () => {
    const input = `Here's a simple footnote[^1]

[^1]: meaningful!

`;

    const { options, markdown } = prepare(input);
    const sut = new MarkdownProcessor(utilsInstance);

    const result = JSON.stringify(sut.process(markdown, options));
    return expect(result).toMatchSnapshot();
});

test('Basic Markdown Syntax > Footnotes in Tables', () => {
    const input = `| Table     | Col 1 | Col 2 | Col 3 |
    | --------- | ----- | ----- | ----- |
    | Footnotes | [^1]  | [^2]  | [^3]  |

    [^1]: fn1
    [^2]: fn2
    [^3]: fn3
`;

    const { options, markdown } = prepare(input);
    const sut = new MarkdownProcessor(utilsInstance);

    const result = JSON.stringify(sut.process(markdown, options));
    return expect(result).toMatchSnapshot();
});

test('Basic Markdown Syntax > Math', () => {
    // escape literals in javascript interpreted string
    const input = readFileSync('test/fixtures/mathjax.md', 'utf8');

    const { options, markdown } = prepare(input);
    const sut = new MarkdownProcessor(utilsInstance);

    const result = JSON.stringify(sut.process(markdown, options));
    return expect(result).toMatchSnapshot();
});

test('Basic Markdown Syntax > Mermaid', () => {
    const input = `---
theme: beige
highlightTheme: css/vs2015.css

---

\`\`\`mermaid
sequenceDiagram
    Alice->>+John: Hello John, how are you?
    Alice->>+John: John, can you hear me?
    John-->>-Alice: Hi Alice, I can hear you!
    John-->>-Alice: I feel great!
\`\`\``;

    const { options, markdown } = prepare(input);
    const sut = new MarkdownProcessor(utilsInstance);

    const result = JSON.stringify(sut.process(markdown, options));
    return expect(result).toMatchSnapshot();
});

test('Basic Markdown Syntax > Callouts', () => {
    const input = `> [!tip] This tip has a custom Header
    > This is an Tip with a custom Title
    <!-- element style="width:40%"-->

    > [!warning]
    > This warning is aligned left
    <!-- element style="width:20%;font-size:24px" align="left"-->

    <br>

    > [!error]
    > Callouts support manipulation through annotations
    <!-- element style="width:20%;font-size:24px" rotate="15"-->

    <grid drag="30 30" drop="11 68" style="font-size:36px">
    > [!example]
    > Here is an Example for an Callout in a Slide. Callouts support dark and white backgrounds and could be sized by annotations
    </grid>

    <grid drag="50 100" drop="50 0" bg="white" pad="50px" style="font-size:36px">
    > [!cite]
    > Here is an Cite for an Callout in a Slide. Callouts support dark and white backgrounds and could be sized by annotations
    </grid>`;

    const { options, markdown } = prepare(input);
    const sut = new MarkdownProcessor(utilsInstance);

    const result = JSON.stringify(sut.process(markdown, options));
    return expect(result).toMatchSnapshot();
});
