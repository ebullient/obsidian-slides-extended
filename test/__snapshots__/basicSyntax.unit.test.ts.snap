// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Markdown Syntax > Blockquotes 1`] = `""<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\n> Human beings face ever more complex and urgent problems, and their effectiveness in dealing with these problems is a matter that is critical to the stability and continued progress of society.\\n\\n- Doug Engelbart, 1961\\n</div>""`;

exports[`Basic Markdown Syntax > Callouts 1`] = `""<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\n<div class=\\"callout callout-color2\\">\\n<div class=\\"callout-title\\">\\n<div class=\\"callout-icon\\">\\n\\n<i class=\\"fas fa-fire-alt\\" ></i>\\n\\n\\n</div>\\n<div class=\\"callout-title-inner\\">\\n\\nThis tip has a custom Header\\n\\n</div>\\n</div>\\n<div class=\\"callout-content\\">\\n\\nThis is an Tip with a custom Title\\n\\n</div>\\n</div>\\n    <!-- .element: style=\\"width: 40%\\" -->\\n\\n<div class=\\"callout callout-color5\\">\\n<div class=\\"callout-title\\">\\n<div class=\\"callout-icon\\">\\n\\n<i class=\\"fas fa-exclamation-triangle\\" ></i>\\n\\n\\n</div>\\n<div class=\\"callout-title-inner\\">\\n\\nWarning\\n\\n</div>\\n</div>\\n<div class=\\"callout-content\\">\\n\\nThis warning is aligned left\\n\\n</div>\\n</div>\\n    <!-- .element: style=\\"width: 20%; font-size: 24px; display: flex; flex-direction: column; align-items: flex-start; justify-content: space-evenly\\" align=\\"left\\" -->\\n\\n    <br>\\n\\n<div class=\\"callout callout-color7\\">\\n<div class=\\"callout-title\\">\\n<div class=\\"callout-icon\\">\\n\\n<i class=\\"fas fa-bolt\\" ></i>\\n\\n\\n</div>\\n<div class=\\"callout-title-inner\\">\\n\\nError\\n\\n</div>\\n</div>\\n<div class=\\"callout-content\\">\\n\\nCallouts support manipulation through annotations\\n\\n</div>\\n</div>\\n    <!-- .element: style=\\"width: 20%; font-size: 24px; transform: rotate(15deg)\\" -->\\n\\n    <div class=\\"\\" style=\\"font-size: 36px; position: absolute; left: 11%; top: 68%; height: 30%; width: 30%; display: flex; flex-direction: column; align-items: center; justify-content: center\\" >\\n\\n<div class=\\"callout callout-color8\\">\\n<div class=\\"callout-title\\">\\n<div class=\\"callout-icon\\">\\n\\n<i class=\\"fas fa-list\\" ></i>\\n\\n\\n</div>\\n<div class=\\"callout-title-inner\\">\\n\\nExample\\n\\n</div>\\n</div>\\n<div class=\\"callout-content\\">\\n\\nHere is an Example for an Callout in a Slide. Callouts support dark and white backgrounds and could be sized by annotations\\n\\n</div>\\n</div>\\n    </div>\\n\\n    <div class=\\"has-light-background\\" style=\\"font-size: 36px; background-color: white; padding: 50px; box-sizing: border-box; position: absolute; left: 50%; top: 0%; height: 100%; width: 50%; display: flex; flex-direction: column; align-items: center; justify-content: center\\" >\\n\\n<div class=\\"callout callout-color9\\">\\n<div class=\\"callout-title\\">\\n<div class=\\"callout-icon\\">\\n\\n<i class=\\"fas fa-quote-left\\" ></i>\\n\\n\\n</div>\\n<div class=\\"callout-title-inner\\">\\n\\nCite\\n\\n</div>\\n</div>\\n<div class=\\"callout-content\\">\\n\\nHere is an Cite for an Callout in a Slide. Callouts support dark and white backgrounds and could be sized by annotations\\n\\n</div>\\n</div>\\n    </div>\\n</div>""`;

exports[`Basic Markdown Syntax > Embeds 1`] = `""<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\n<!-- begin::[Obsidian Notes.md] -->\\n# Notes about Obsidian\\nA knowledge base that works on local Markdown files\\n\\n# Link\\nLink to Obsidian Homepage: http://obsidian.md\\n<!-- end::[Obsidian Notes.md] -->\\n</div>\\n---\\n<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\n<!-- begin::[Obsidian Notes.md] -->\\nLink to Obsidian Homepage: http://obsidian.md\\n<!-- end::[Obsidian Notes.md] -->\\n</div>\\n---\\n<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\n<!-- begin::[Obsidian Notes.md] -->\\n# Notes about Obsidian\\nA knowledge base that works on local Markdown files\\n\\n# Link\\nLink to Obsidian Homepage: http://obsidian.md\\n<!-- end::[Obsidian Notes.md] -->\\n</div>\\n---\\n<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\n![](Obsidian%20Notes.md#Link)\\n</div>""`;

exports[`Basic Markdown Syntax > Footnotes 1`] = `""<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\nHere's a simple footnote<sup id=\\"fnref:1\\" role=\\"doc-noteref\\">1</sup>\\n\\n\\n\\n\\n<div class=\\"footnotes\\" role=\\"doc-endnotes\\">\\n<ol>\\n<li id=\\"fn:1\\" role=\\"doc-endnote\\" class=\\"footnote\\"><p>\\n\\nmeaningful!\\n\\n</p></li></ol>\\n</div>\\n</div>""`;

exports[`Basic Markdown Syntax > Footnotes in Tables 1`] = `""<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\n| Table     | Col 1 | Col 2 | Col 3 |\\n    | --------- | ----- | ----- | ----- |\\n| Footnotes |<sup id=\\"fnref:1\\" role=\\"doc-noteref\\">1</sup>|<sup id=\\"fnref:2\\" role=\\"doc-noteref\\">2</sup>|<sup id=\\"fnref:3\\" role=\\"doc-noteref\\">3</sup>|\\n\\n\\n\\n\\n\\n<div class=\\"footnotes\\" role=\\"doc-endnotes\\">\\n<ol>\\n<li id=\\"fn:1\\" role=\\"doc-endnote\\" class=\\"footnote\\"><p>\\n\\nfn1\\n\\n</p></li><li id=\\"fn:2\\" role=\\"doc-endnote\\" class=\\"footnote\\"><p>\\n\\nfn2\\n\\n</p></li><li id=\\"fn:3\\" role=\\"doc-endnote\\" class=\\"footnote\\"><p>\\n\\nfn3\\n\\n</p></li></ol>\\n</div>\\n</div>""`;

exports[`Basic Markdown Syntax > Headers 1`] = `""<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\n# This is a heading 1\\n## This is a heading 2\\n### This is a heading 3\\n#### This is a heading 4\\n##### This is a heading 5\\n###### This is a heading 6\\n</div>""`;

exports[`Basic Markdown Syntax > Images 1`] = `""<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\nInsert image with standard markdown syntax\\n\\n<img src=\\"https://picsum.photos/id/1006/500/300\\" alt=\\"Image\\" style=\\"object-fit: scale-down\\">\\n</div>\\n---\\n<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\nInsert image that lies outside of your vault\\n\\n<img src=\\"/local-file-url/Users/testUser/Desktop/howToUse.png\\" alt=\\"Image\\" style=\\"object-fit: scale-down\\">\\n</div>\\n---\\n<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\nInsert image with obsidian markdown syntax\\n\\n<img src=\\"/documentation/Image.jpg\\" alt=\\"\\" style=\\"object-fit: scale-down\\">\\n</div>\\n---\\n<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\nScale image to a width of 100 px\\n\\n<img src=\\"/documentation/Image.jpg\\" alt=\\"\\" style=\\"width: 100px; object-fit: fill\\">\\n</div>\\n---\\n<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\nScale image to a width of 300x100 px\\n\\n<img src=\\"/documentation/Image.jpg\\" alt=\\"\\" style=\\"object-fit: cover; width: 300px; height: 100px\\">\\n</div>""`;

exports[`Basic Markdown Syntax > Inline Code 1`] = `""<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\n### Text inside \`backticks\` on a line will be formatted like code.\\n</div>""`;

exports[`Basic Markdown Syntax > Links 1`] = `""<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\nExternal Links\\n\\nhttp://obsidian.md - automatic!\\n\\n[Obsidian](http://obsidian.md)\\n</div>\\n---\\n<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\nObsidian URI links\\n\\n[Link to note](obsidian://open?path=D:%2Fpath%2Fto%2Ffile.md)\\n\\n[Link to note](obsidian://open?vault=MainVault&file=MyNote.md)\\n</div>\\n---\\n<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\nInternal Links\\n\\nThis Internal link will be displayed as normal text\\n\\nThis Link will use its alias for displaying\\n</div>""`;

exports[`Basic Markdown Syntax > Links 2`] = `""<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\nInternal Links\\n\\nThis [Internal link](obsidian://open?vault=test-vault&file=Internal%20link) will be displayed as normal text\\n\\nThis [Link](obsidian://open?vault=test-vault&file=Internal%20link) will use its alias for displaying\\n</div>""`;

exports[`Basic Markdown Syntax > Lists 1`] = `""<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\n- Item 1\\n- Item 2\\n\\t- Item 2a\\n\\t- Item 2b\\n</div>\\n---\\n<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\n1. Item 1\\n1. Item 2\\n1. Item 3\\n   1. Item 3a\\n   1. Item 3b\\n</div>""`;

exports[`Basic Markdown Syntax > Math 1`] = `""<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\n$$\\\\begin{vmatrix}a & b\\\\\\\\\\\\c & d\\\\end{vmatrix}=ad-bc$$\\n\\nYou can also do inline math like $s^{-2}\\\\_{n}\\\\sum\\\\_{i=1}^{n}$\\n</div>""`;

exports[`Basic Markdown Syntax > Mermaid 1`] = `""<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\n<div class=\\"mermaid\\">\\nsequenceDiagram\\n    Alice->>+John: Hello John, how are you?\\n    Alice->>+John: John, can you hear me?\\n    John-->>-Alice: Hi Alice, I can hear you!\\n    John-->>-Alice: I feel great!\\n\\n</div>\\n</div>""`;

exports[`Basic Markdown Syntax > Tables 1`] = `""<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\nFirst Header | Second Header\\n------------ | ------------\\nContent from cell 1 | Content from cell 2\\nContent in the first column | Content in the second column\\n</div>""`;

exports[`Basic Markdown Syntax > Text style 1`] = `""<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\n*This text will be italic*\\n\\n_This will also be italic_\\n\\n**This text will be bold**\\n\\n**This will also be bold**\\n\\n\\n\\n_You **can** combine them_\\n</div>\\n---\\n<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\nAny word wrapped with two tildes (like ~~this~~) will appear crossed out.\\n\\nAny word wrapped with two equal signs (like <mark>this</mark>) will appear as highlighted.\\n</div>""`;

exports[`Basic Markdown Syntax > Videos 1`] = `""<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\nInsert image that lies outside of your vault\\n\\n<video style=\\"object-fit: scale-down\\" controls=\\"\\"><source src=\\"/local-file-url/Users/testUser/Desktop/video.mp4\\" alt=\\"Video\\" type=\\"video/mp4\\" /></video>\\n</div>\\n---\\n<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\nInsert image with obsidian markdown syntax\\n\\n<video style=\\"object-fit: scale-down\\" controls=\\"\\"><source src=\\"/documentation/video.mp4\\" alt=\\"\\" type=\\"video/mp4\\" /></video>\\n</div>\\n---\\n<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\nScale image to a width of 100 px\\n\\n<video style=\\"width: 100px; object-fit: fill\\" controls=\\"\\"><source src=\\"/documentation/video.mp4\\" alt=\\"\\" type=\\"video/mp4\\" /></video>\\n</div>\\n---\\n<!-- .slide: class=\\"drop\\" -->\\n<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">\\n\\nScale image to a width of 300x100 px, disable controls\\n\\n<video style=\\"object-fit: cover; width: 300px; height: 100px\\"><source src=\\"/documentation/video.mp4\\" alt=\\"\\" type=\\"video/mp4\\" /></video>\\n</div>""`;
